### DISCUSSION TITLE
Ability to anonymise user/order/customer data

### DISCUSSION THREAD

**Discussion by peteeveleigh** [craftcms/commerce#2453] [Ideas]
**What are you trying to do?**
When submitting support requests we are often asked to provide a database dump. We can't send databases containing customer data due to GDPR and other privacy concerns. There's no built in "safe" way to delete this data.

Also bear in mind we may have tens of thousands of users, customers and orders.

**What's your proposed solution?**
Either a system to remove all user, customers, and orders OR a means to anonymise them by replacing names, emails, and address data. I can see anonymisation being tricky if there are custom fields on an order or user records. 

Ideally a system that allowed the developer to choose to which fields to anonymise/delete/replace. But we appreciate that's a non-trivial job!

**Comment by lukeholder** [comment]
Would the reset data console command work for you?

```
$ php craft commerce/reset-data

Resetting Commerce data will permanently delete all orders, subscriptions, payment sources, customers, addresses and reset discount usages ... do you wish to continue? [no] 
```

Was added in 3.2.8

**Comment by lukeholder** [comment]
Can keep this open as an anonymise feature request for a related console command.

**Comment by peteeveleigh** [comment]
Oh that command might just do it!

**Comment by lukeholder** [comment]
No worries, change the feature request to 

> Ability to anonymise user/order/customer data

Will keep it on the list.
